### PR TITLE
Bump `kind` version

### LINKED ISSUE #431: Bump kind version

### PR DESCRIPTION
**Pull request by embano1**
Fixes #431

## Proposed Changes

- :broom: Bump `kind` version
-
-

### Pre-review Checklist

- [ ] **At least 80% unit test coverage**
- [ ] **E2E tests** for any new behavior
- [ ] **Docs** for any user-facing impact

**Release Note**

```release-note

```

### CODE REVIEW

File: .github/workflows/kind-e2e.yaml [hunk 1]
```diff
@@ -34,7 +34,7 @@ jobs:
     - name: Setup KinD Cluster with VMware sources.
       working-directory: ./src/github.com/${{ github.repository }}
       env:
-        KIND_VERSION: v0.11.1
+        KIND_VERSION: v0.12.0
         NODE_VERSION: v1.23.5
```
Comment (kvmware) [inline .github/workflows/kind-e2e.yaml:38]: I don't know if that version of NODE is built for 0.13 of KIND https://github.com/kubernetes-sigs/kind/releases/tag/v0.12.0 and `The default node image is a Kubernetes v1.23.4 image` so might not even need that change that got merged (which was what I was trying to hint at :))

Comment (embano1) [reply .github/workflows/kind-e2e.yaml:38]: Ah sorry, misunderstood you then. 

On one hand, being explicit about the version helps IMHO. But I'm also happy to change to using `v1.23.4` to align with `kind` (though this version will change with `kind` release and so we need to keep them in sync). Thoughts?

Comment (kvmware) [reply .github/workflows/kind-e2e.yaml:38]: Yep, if someone bumps KIND version in future will have to be sure that the NODE version aligns or is supported. If there is not a particular reason to pin the NODE version, I would just remove the NODE version and just bump KIND

Comment (embano1) [reply .github/workflows/kind-e2e.yaml:38]: That makes sense!

**Review by krsna-m — APPROVED** [review]

**Review by gabo1208 — APPROVED** [review]

**Pull request merged**